bots/notify/src/main/java/org/openjdk/skara/bots/notify/issue/IssueNotifier.java

```diff
@@ -305,7 +305,18 @@ public void onNewCommits(HostedRepository repository, Repository localRepository
                             var existing = Backports.findIssue(issue, fixVersion);
                             if (existing.isEmpty()) {
                                 log.info("Creating new backport for " + issue.id() + " with fixVersion " + requestedVersion);
-                                issue = jbsBackport.createBackport(issue, requestedVersion, username.orElse(null), defaultSecurity(branch));
+                                try {
+                                    issue = jbsBackport.createBackport(issue, requestedVersion, username.orElse(null), defaultSecurity(branch));
+                                } catch (Exception exp) {
```

Comment:
Catching `Exception` is too wide a net. We need to catch `UncheckedRestException` and ideally check that the response code indicates this kind of error. Unfortunately I don't know what response that would be, so perhaps it's ok to just change the exception type for now.

Exception variables in catch statements are usually just called `e` in this codebase.

Reply:
Yes, I agree, used the more specific exception as suggested.

Regarding the specific exception code, yes I tried to run two test cases in parallel by putting the debug points to simulate the race condition but TestHost returned 2nd issue as a new one. I guess because of two instances of TestHost (one per test case). So could not verify the exception code.

Reply:
If you are interested, I think you can add the status code to the log with method `UncheckedRestException#getStatusCode`.

Reply:
I think logging the statusCode as suggested by Zhao is a good idea.

Reply:
Or never mind, we already log when throwing that exception.

Reply:
I don't think so, when the race condition occurs, the exception won't be thrown again.

Reply:
We log in `RestRequest::transformBadResponse` before throwing the exception.

Reply:
Oh, right.